glue/sales_val_flagging.py

```diff
@@ -786,7 +786,14 @@ def write_to_table(df, table_name, s3_warehouse_bucket_path, run_id):
     if filtered_df.sv_outlier_type.isna().sum() == 0:
         print("WARNING: No new sales to flag")
     else:
-        df = df.astype({col[0]: sql_type_to_pd_type(col[1]) for col in metadata})
+        # Make sure None types aren't utilized in type conversion
+        conversion_dict = {
+            col[0]: sql_type_to_pd_type(col[1])
+            for col in metadata
+            if sql_type_to_pd_type(col[1]) is not None
+        }
+        df = df.astype(conversion_dict)
+
```

Comment:
This function broke because of a meta data change from athena. I'm assuming from awswragler. This stops the conversion of dtypes to `Null` which threw an error.

Reply:
**[Question, non-blocking]** What are the cases where we were finding the SQL type code in the cursor description for a column to be null?

Reply:
My best guess is that the method functionality of `as.type()` on `pd.DataFrame` objects changed between pandas versions. Prior to the package change, I did not need to specify the `if sql_type_to_pd_type(col[1]) is not None`, it was skipping over those cases. Now, the dict comprehension seems to actually attempt to make a dtype change for each column.

The only change in the `sql_type_to_pd_type()` function is the following:

```
    if sql_type in ["decimal"]:
        return "float64"
```

So for non "decimal" types, the dict comprehension still attempts to assign a change, which is null because the dtype of the column doesn't exist in the `sql_type_to_pd_type()` function. This is my best go at understanding the error @jeancochrane

Reply:
If you're confident that this results in no output changes then I'm satisfied with this solution @wagnerlmichael, even if it's a bit mysterious.